example/src/main/java/io/radar/example/MainActivity.kt

```diff
@@ -5,29 +5,56 @@ import android.location.Location
 import android.os.Build
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.os.PersistableBundle
 import android.util.Log
-import androidx.core.app.ActivityCompat
+import android.widget.Button
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
 import io.radar.sdk.Radar
 import io.radar.sdk.RadarTrackingOptions
 import io.radar.sdk.RadarTripOptions
+import org.json.JSONObject
 import java.util.*
 
 class MainActivity : AppCompatActivity() {
 
+    private lateinit var requestLocationPermissionLauncher: ActivityResultLauncher<Array<String>>
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
-                ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.BLUETOOTH_SCAN, Manifest.permission.BLUETOOTH_CONNECT, Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_BACKGROUND_LOCATION), 0)
+        requestLocationPermissionLauncher = registerForActivityResult(
+            ActivityResultContracts.RequestMultiplePermissions()
+        ) { isGrantedMap ->
+            // Check if the requested permission was granted
+            if (isGrantedMap.all { it.value == true }) {
+                runDemo()
             } else {
-                ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.BLUETOOTH_ADMIN, Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_BACKGROUND_LOCATION), 0)
+                Log.e("example", "Requires all location permissions to be granted")
```

Comment:
are all permissions really required? or is it just the Android equivalent of when in use?

Reply:
This is for the Example app, which probably doesn't have to be nuanced enough to support foreground-only vs foreground and background right now.